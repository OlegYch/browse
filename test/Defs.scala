package defs

object Defs
{
	class A
	class B[T] {
		def x(t: T): T = t
	}
	class C[T <: Y, Y]
	class D[T](t: T)
	class E extends B[Int]

	trait F
	trait G extends F
	class H extends A with G
	class I extends F

	private[defs] final class J private(val i: Int) extends H with G

	lazy val x = 3
	lazy val xx: Int = 3

	def q[T](t: T) = t
	def qq[T](t: T): List[T] = t :: Nil

	private[this] def z: Int = (new E).x(3)
}