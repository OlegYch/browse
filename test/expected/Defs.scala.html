<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
    <head>
        <meta http-equiv="Content-Type" content="text/html;charset=utf-8" ></meta>
        <title>Defs.scala</title>
        <script type="text/javascript" src="jquery-all.js"></script>
        <script type="text/javascript" src="linked.js"></script>
        <link rel="stylesheet" type="text/css" href="style.css" title="Style"></link>
    </head>
    <body>
        <pre>
<span class="keyword">package</span> defs

<span class="keyword">object</span> <a title="defs.Defs.type" id="6942">Defs</a>
<a href="#6943" title="defs.Defs.type" class="delimiter">{</a>
	<span class="keyword">class</span> <a title="class A extends AnyRef" id="16331">A</a>
	<span class="keyword">class</span> <a title="class B[T] extends AnyRef" id="16332">B</a><span class="delimiter">[</span><a title="" id="16333">T</a><span class="delimiter">]</span> <a href="#16332" title="defs.Defs.B[T]" class="delimiter">{</a>
		<span class="keyword">def</span> <a title="(t: T)T" id="16360">x</a><span class="delimiter">(</span><a title="T" id="16362">t</a>: <a href="#16333" title="T">T</a><span class="delimiter">)</span>: <a href="#16333" title="T">T</a> = <a href="#16362" title="T">t</a>
	<span class="delimiter">}</span>
	<span class="keyword">class</span> <a title="class C[T &lt;: Y, Y] extends AnyRef" id="16334">C</a><span class="delimiter">[</span><a title=" &lt;: Y" id="16335">T</a> &lt;: Y, <a title="" id="16336">Y</a><span class="delimiter">]</span>
	<span class="keyword">class</span> <a title="class D[T] extends AnyRef" id="16337">D</a><span class="delimiter">[</span><a title="" id="16338">T</a><span class="delimiter">]</span><a href="#16337" title="defs.Defs.D[T]" class="delimiter">(</a><a title="T" id="16370">t</a>: <a href="#16338" title="T">T</a><span class="delimiter">)</span>
	<span class="keyword">class</span> <a title="class E extends defs.Defs.B[Int]" id="16339">E</a> <a href="#16339" title="defs.Defs.E" class="keyword">extends</a> <a href="#16332" title="defs.Defs.B[Int]">B</a><span class="delimiter">[</span>Int<span class="delimiter">]</span>

	<span class="keyword">trait</span> <a title="trait F extends AnyRef" id="16340">F</a>
	<span class="keyword">trait</span> <a title="trait G extends AnyRef with defs.Defs.F" id="16341">G</a> <span class="keyword">extends</span> <a href="#16340" title="defs.Defs.F">F</a>
	<span class="keyword">class</span> <a title="class H extends defs.Defs.A with defs.Defs.G" id="16342">H</a> <a href="#16342" title="defs.Defs.H" class="keyword">extends</a> <a href="#16331" title="defs.Defs.A">A</a> <span class="keyword">with</span> <a href="#16341" title="defs.Defs.G">G</a>
	<span class="keyword">class</span> <a title="class I extends AnyRef with defs.Defs.F" id="16343">I</a> <a href="#16343" title="defs.Defs.I" class="keyword">extends</a> <a href="#16340" title="defs.Defs.F">F</a>

	<span class="keyword">private</span><span class="delimiter">[</span>defs<span class="delimiter">]</span> <span class="keyword">final</span> <span class="keyword">class</span> <a title="class J extends defs.Defs.H with defs.Defs.G" id="16344">J</a> <a href="#16344" title="defs.Defs.J" class="keyword">private</a><span class="delimiter">(</span><span class="keyword">val</span> <a title="Int" id="16385">i</a>: <span title="Int">Int</span><span class="delimiter">)</span> <span class="keyword">extends</span> <a href="#16342" title="defs.Defs.H">H</a> <span class="keyword">with</span> <a href="#16341" title="defs.Defs.G">G</a>

	<span class="keyword">lazy</span> <span class="keyword">val</span> <a title="Int" id="16345">x</a> = <span title="Int(3)" class="int">3</span>
	<span class="keyword">lazy</span> <span class="keyword">val</span> <a title="Int" id="16347">xx</a>: <span title="Int">Int</span> = <span title="Int(3)" class="int">3</span>

	<span class="keyword">def</span> <a title="[T](t: T)T" id="16349">q</a><span class="delimiter">[</span><a title="" id="16351">T</a><span class="delimiter">]</span><span class="delimiter">(</span><a title="T" id="16386">t</a>: <a href="#16351" title="T">T</a><span class="delimiter">)</span> = <a href="#16386" title="T">t</a>
	<span class="keyword">def</span> <a title="[T](t: T)List[T]" id="16352">qq</a><span class="delimiter">[</span><a title="" id="16354">T</a><span class="delimiter">]</span><span class="delimiter">(</span><a title="T" id="16388">t</a>: <a href="#16354" title="T">T</a><span class="delimiter">)</span>: <span title="List[T]">List</span><span class="delimiter">[</span>T<span class="delimiter">]</span> = <a href="#16388" title="T">t</a> <a href="#16390" title="(x: T)List[T]">::</a> <span title="scala.collection.immutable.Nil.type">Nil</span>

	<span class="keyword">private</span><span class="delimiter">[</span><span class="keyword">this</span><span class="delimiter">]</span> <span class="keyword">def</span> <a title="=&gt; Int" id="16355">z</a>: <span title="Int">Int</span> = <span class="delimiter">(</span><span title="defs.Defs.E" class="keyword">new</span> <a href="#16339" title="defs.Defs.E">E</a><span class="delimiter">)</span>.<a href="#16360" title="(t: Int)Int">x</a><span class="delimiter">(</span><span title="Int(3)" class="int">3</span><span class="delimiter">)</span>
<span class="delimiter">}</span>
        </pre>
    </body>
</html>
